bundle/src/main/java/com/adobe/acs/commons/redirects/filter/RedirectFilter.java

```diff
@@ -0,0 +1,508 @@
+import static org.osgi.framework.Constants.SERVICE_RANKING;
+
+/**
+ * A request filter that implements support for virtual redirects.
+ */
+@Component(service = {Filter.class, RedirectFilterMBean.class, EventHandler.class}, property = {
+        SERVICE_DESCRIPTION + "=A request filter implementing support for virtual redirects",
+        SLING_FILTER_SCOPE + "=" + EngineConstants.FILTER_SCOPE_REQUEST,
+        SERVICE_RANKING + ":Integer=10000",
+        "jmx.objectname=" + "com.adobe.acs.commons:type=Redirect Manager",
+        EventConstants.EVENT_TOPIC + "=" + ReplicationAction.EVENT_TOPIC,
+        ResourceChangeListener.CHANGES + "={ADDED, REMOVED, CHANGED}"
+
+})
+@Designate(ocd = RedirectFilter.Configuration.class)
+public class RedirectFilter extends AnnotatedStandardMBean
+        implements Filter, EventHandler, ResourceChangeListener, RedirectFilterMBean {
+
+    public static final String DEFAULT_STORAGE_PATH = "/var/acs-commons/redirects";
+    public static final String ACS_REDIRECTS_RESOURCE_TYPE = "acs-commons/components/utilities/manage-redirects";
+    public static final String REDIRECT_RULE_RESOURCE_TYPE = ACS_REDIRECTS_RESOURCE_TYPE + "/redirect-row";
+
+
+    private static final Logger log = LoggerFactory.getLogger(MethodHandles.lookup().lookupClass());
+
+    static final String SERVICE_NAME = "redirect-manager";
+
+    @ObjectClassDefinition(name = "ACS Commons Redirect Filter")
+    public @interface Configuration {
+        @AttributeDefinition(name = "Enable Redirect Filter", description = "Indicates whether the redirect filter is enabled or not.", type = AttributeType.BOOLEAN)
+        boolean enabled() default true;
+
+        @AttributeDefinition(name = "Rewrite Location Header", description = "Apply Sling Resource Mappings (/etc/map) to Location header. " +
+                "Use if Location header should rewritten using ResourceResolver#map", type = AttributeType.BOOLEAN)
+        boolean rewriteUrls() default true;
+
+        @AttributeDefinition(name = "Request Extensions", description = "List of extensions for which redirection is allowed", type = AttributeType.STRING)
+        String[] extensions() default {};
+
+        @AttributeDefinition(name = "Request Paths", description = "List of paths for which redirection is allowed", type = AttributeType.STRING)
+        String[] paths() default {"/content"};
+
+        @AttributeDefinition(name = "Preserve Query String", description = "Preserve query string in redirects", type = AttributeType.BOOLEAN)
+        boolean preserveQueryString() default true;
+
+        @AttributeDefinition(name = "Storage Path", description = "The path in the repository to store redirect configurations", type = AttributeType.STRING)
+        String storagePath() default DEFAULT_STORAGE_PATH;
+
+        @AttributeDefinition(name = "On Delivery Headers", description = "Optional On-Delivery headers in the name:value format,"+

```

Comment:
"Additional redirect response headers"?

On Delivery is a bit vague for me....

Reply:
renamed to  additionalHeaders()
```java
        @AttributeDefinition(name = "Additional Response Headers", description = "Optional response headers in the name:value format to apply on delivery,"
                + " e.g. Cache-Control: max-age=3600", type = AttributeType.STRING)
        String[] additionalHeaders() default {};
```